File: typescript/src/models/riskSignal.ts
// this file is @generated
import { type RiskSignalAction, RiskSignalActionSerializer } from "./riskSignalAction";
import { type RiskSignalType, RiskSignalTypeSerializer } from "./riskSignalType";

export interface RiskSignal {
  type: RiskSignalType;
  score: number;
  action: RiskSignalAction;
}

export const RiskSignalSerializer = {
  _fromJsonObject(object: any): RiskSignal {
    return {
      type: RiskSignalTypeSerializer._fromJsonObject(object["type"]),
      score: object["score"],
      action: RiskSignalActionSerializer._fromJsonObject(object["action"]),
    };
  },

  _toJsonObject(self: RiskSignal): any {
    return {
      type: RiskSignalTypeSerializer._toJsonObject(self.type),
      score: self.score,
      action: RiskSignalActionSerializer._toJsonObject(self.action),
    };
  },
};

File: rust/src/models/risk_signal.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::{risk_signal_action::RiskSignalAction, risk_signal_type::RiskSignalType};

#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
pub struct RiskSignal {
    pub r#type: RiskSignalType,
    pub score: i32,
    pub action: RiskSignalAction,
}
